core/audits/prioritize-lcp-image.js

```diff
@@ -131,7 +133,31 @@ class PrioritizeLcpImage extends Audit {
     // Get last candidate, in case there was more than one.
     }).sort((a, b) => b.ts - a.ts)[0];
 
-    return lcpImagePaintEvent?.args.data?.imageUrl;
+    const lcpUrl = lcpImagePaintEvent?.args.data?.imageUrl;
+    if (!lcpUrl) return;
+
+    const candidates = networkRecords.filter(record => {
+      return record.url === lcpUrl &&
+          record.finished &&
+          // Same frame as LCP trace event.
+          record.frameId === lcpImagePaintEvent.args.frame &&
+          record.networkRequestTime < (processedNavigation.timestamps.largestContentfulPaint || 0);
+    }).map(record => {
+      // Follow any redirects to find the real image request.
+      while (record.redirectDestination) {
+        record = record.redirectDestination;
+      }
+      return record;
+    }).filter(record => {
+      // Don't select if also loaded by some other means (xhr, etc). `resourceType`
+      // isn't set on redirect _sources_, so have to check after following redirects.
+      return record.resourceType === 'Image';
+    });
+
+    // If there are still multiple candidates, at this point it appears the page
+    // simply made multiple requests for the image. The first loaded is the best
+    // guess of the request that made the image available for use.
+    return candidates.sort((a, b) => a.networkEndTime - b.networkEndTime)[0];
```

Comment:
Out of order network requests are not something I've actually observed, but `NetworkRecords` are ordered by the order `requestWillBeSent` messages come in, so possibly, in theory, they could be out of order, and it probably makes more sense to look at the first finished request rather than the first one to start.

It doesn't feel like the renderer and network stack would let that occur in any way that really matters for image requests in particular, but better to be sure.